src/electron/platform/android/appium-service-configurator.ts

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import ADB from 'appium-adb';
+import {
+    AndroidServiceConfigurator,
+    DeviceInfo,
+    PackageInfo,
+    PermissionInfo,
+} from 'electron/platform/android/android-service-configurator';
+
+type AdbDevice = {
+    udid: string;
+};
+
+const servicePackageName: string = 'com.microsoft.accessibilityinsightsforandroidservice';
+
+export class AppiumServiceConfigurator implements AndroidServiceConfigurator {
+    private readonly adb: ADB;
+
+    constructor(adb: ADB) {
+        this.adb = adb;
+    }
+
+    public getConnectedDevices = (): Promise<Array<DeviceInfo>> => {
+        return new Promise<Array<DeviceInfo>>(async (resolve, reject) => {
+            try {
+                // First add devices that are flagged as emulator, then
+                // add devices that aren't flagged as emulators
+                const detectedDevices = {};
+                await this.AddDevices(
+                    await this.adb.getConnectedEmulators(),
+                    true,
+                    detectedDevices,
+                );
+                await this.AddDevices(await this.adb.getConnectedDevices(), false, detectedDevices);
+
+                const output: Array<DeviceInfo> = [];
+                Object.values(detectedDevices).forEach(device => output.push(device as DeviceInfo));
+                resolve(output);
+            } catch (error) {
+                reject(error);
+            }
+        });
+    };
+
+    private AddDevices(
```

Comment:
We usually prefer `camelCase` to `PascalCase` for functions

Reply:
Yup, muscle memory from C#